src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -0,0 +1,440 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.nanos.approval',
+  name: 'ApprovalRequest',
+  documentation: 'Approval requests are stored in approvalRequestDAO and' +
+  'represent a single approval request for a single user.',
+
+  implements: [
+    'foam.nanos.auth.CreatedAware',
+    'foam.nanos.auth.CreatedByAware',
+    'foam.nanos.auth.LastModifiedAware',
+    'foam.nanos.auth.LastModifiedByAware'
+  ],
+
+  javaImports: [
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.logger.Logger'
+  ],
+
+  requires: [
+    'foam.dao.AbstractDAO',
+    'foam.nanos.approval.ApprovalStatus'
+  ],
+
+  imports: [
+    'approvalRequestDAO',
+    'ctrl'
+  ],
+
+  tableColumns: [
+    'id',
+    'description',
+    'classification',
+    'objId',
+    'viewReference',
+    'approver',
+    'status',
+    'memo',
+    'approve',
+    'reject'
+  ],
+
+  sections: [
+    {
+      name: 'basicInformation',
+      isAvailable: () => false
```

Comment:
Can this be permissioned instead?

Reply:
Was last minute fix, just changed it to permissionRequired: true

Reply:
Was last minute fix, just changed it to permissionRequired: true

Reply:
Was last minute fix, just changed it to permissionRequired: true